Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (x_pos : 0 < PGame.mk l r L R)
+    (i'' : InvTy {i // 0 < L i} r b) :=
+  match i'' with
+  | InvTy.zero => (0, 0, 0)
+  | InvTy.left₁ i j =>
+    (mk (R i) (h5.2.2 i), mk (R i).inv' (h4 i),
+      mk ((PGame.mk l r L R).inv'.moveLeft j) (by apply inv'_numeric_left <;> tauto))
+  | InvTy.left₂ i j =>
+    (mk (L i) (h5.2.1 i), mk (L i).inv' (h3 i i.2),
+      mk ((PGame.mk l r L R).inv'.moveRight j) (by apply inv'_numeric_right <;> tauto))
+  | InvTy.right₁ i j =>
+    (mk (L i) (h5.2.1 i), mk (L i).inv' (h3 i i.2),
+      mk ((PGame.mk l r L R).inv'.moveLeft j) (by apply inv'_numeric_left <;> tauto))
+  | InvTy.right₂ i j =>
+    (mk (R i) (h5.2.2 i), mk (R i).inv' (h4 i),
+      mk ((PGame.mk l r L R).inv'.moveRight j) (by apply inv'_numeric_right <;> tauto))
+
+/-- The identity `1 - x*y'' = (1 - x * y')*(x' - x)/x'` -/
+lemma eq1 {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < L i → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric) {b : Bool}
+    (x_pos : 0 < PGame.mk l r L R)
+    (inv_l : ∀ (i : { i // 0 < L i }), mk (L i) (h5.2.1 i) * mk (L i).inv' (h3 i i.2) = 1)
+    (inv_r : ∀ j, mk (R j) (h5.2.2 j) * mk (R j).inv' (h4 j) = 1)
+    (i'' : InvTy {i // 0 < L i} r b) :
+    let (x', x'_inv, y') := components h3 h4 h5 x_pos i''
+    let x := mk (PGame.mk l r L R) h5
+    let y'' := match b with
+      | false => mk ((PGame.mk l r L R).inv'.moveLeft i'')
+        (by apply inv'_numeric_left <;> tauto)
+      | true => mk ((PGame.mk l r L R).inv'.moveRight i'')
+        (by apply inv'_numeric_right <;> tauto)
+    match i'' with
+    | InvTy.zero => true
+    | _ => 1 - x*y'' = (1 - x * y')*(x' - x)*x'_inv := by
+  cases i'' <;> simp only [components]
+  all_goals
+    simp only [inv', moveLeft_mk, invVal, moveRight_mk]
+    rw [mk_mul, mk_add, mk_mul, mk_sub]
+    on_goal 2 => first | apply h5.2.1 | apply h5.2.2
+    on_goal 2 => exact h5
+    on_goal 2 => apply invVal_numeric <;> tauto
+    on_goal 2 => exact numeric_one
+    on_goal 2 => case _ j _ => first | exact h3 j j.2 | exact h4 j
+    simp only [← one_def]
+  case' left₁ i j | right₂ i j =>
+    specialize inv_r i
+    have : IsUnit (mk (R i) (h5.2.2 i)) := by rw [isUnit_iff_exists_inv]; exact ⟨_, inv_r⟩
+    lift (mk (R i) (h5.2.2 i)) to Surrealˣ using this with s
+    have : mk (R i).inv' (h4 i) = ↑s⁻¹ := Eq.symm (Units.inv_eq_of_mul_eq_one_right inv_r)
+  case' left₂ i j | right₁ i j =>
+    specialize inv_l i
+    have : IsUnit (mk (L i) (h5.2.1 i)) := by rw [isUnit_iff_exists_inv]; exact ⟨_, inv_l⟩
+    lift (mk (L i) (h5.2.1 i)) to Surrealˣ using this with s
+    have : mk (L i).inv' (h3 i i.2) = ↑s⁻¹ := Eq.symm (Units.inv_eq_of_mul_eq_one_right inv_l)
+  all_goals
+    rw [this]
+    field_simp
+    ring
+
+lemma onag_1_10_i' {l r} {L : l → PGame} {R : r → PGame}
```

Comment:
Most of the same comments on the previous proof apply here. Maybe this could be named `one_between_mul_options`?